common/bufferedpipe/bufferedpipe.go

```diff
@@ -0,0 +1,49 @@
+package bufferedpipe
+
+import (
+	"fmt"
+	"io"
+)
+
+type pipe struct {
+	bufs chan []byte
+	io.Reader
+	closer io.Closer
+}
+
+// New destroys pipes.  No wait, that can't be right.
+func New() io.ReadWriteCloser {
+	r, w := io.Pipe()
+	p := &pipe{
+		bufs:   make(chan []byte, 100),
+		Reader: r,
+		closer: r,
+	}
+	go p.loop(w)
+	return p
+}
```

Comment:
I guess this has slightly different behaviour than `bufio.NewWriter(w)`, correct? In that it's never explicitly flushed...